### PR TITLE
Implemented DAGQL as tree-walk interpreter

### PR DESCRIPTION
**Pull request by CoffeeTableEspresso**

### CODE REVIEW

File: dagql/__main__.py [hunk 1]
```diff
@@ -0,0 +1,31 @@
+from .lexer import Lexer, LexerError
+from .parser import Parser, ParserError
+from .interpreter import Interpreter, InterpreterError
+import sys
+
+def main():
+    compiler = Interpreter()
```
Comment (michaelenglo) [inline dagql/__main__.py:7]: nitpick: not sure why we call interpreter a  compiler here?

File: dagql/ast.py [hunk 2]
```diff
@@ -0,0 +1,89 @@
+class AST:
+    pass
+
+class SelectEdge(AST):
+    def __init__(self, exprs, edge, start, end, subquery, traversal_order):
```
Comment (michaelenglo) [inline dagql/ast.py:5]: If we were to continue extending the project, then maybe I would at add docstrings for explaining each of the attributes.

File: dagql/parser.py [hunk 3]
```diff
@@ -0,0 +1,274 @@
...
+        self.pos = 0
+        self.current_token = self.tokens[0]
+        #self.next_token = self.self.tokens[1]
+    def advance(self):
+        self.pos += 1
+        if self.pos < len(self.tokens):
+            self.current_token = self.tokens[self.pos]
+        else:
+            self.current_token = Token(TokenTypes.EOF, None, self.tokens[-1].line)
+        #self.next_token = self.tokens[self.pos]
```
Comment (michaelenglo) [inline dagql/parser.py:26]: nitpick: remove comments if no longer used.

File: dagql/parser.py [hunk 4]
```diff
@@ -0,0 +1,274 @@
...
+
+class ParserError(Exception):
+    pass
+
+class Parser(object):
+    def __init__(self, tokens):
+        self.tokens = tokens
+        self.pos = 0
+        self.current_token = self.tokens[0]
+        #self.next_token = self.self.tokens[1]
```
Comment (michaelenglo) [inline dagql/parser.py:19]: same here

**Review by michaelenglo — APPROVED** [review]

File: dagql/__main__.py (hunk 1 above)
Comment (CoffeeTableEspresso) [reply dagql/__main__.py:7]: it's based on my old code, where this was a compiler. I forgot to rename the variable.

File: dagql/ast.py (hunk 2 above)
Comment (CoffeeTableEspresso) [reply dagql/ast.py:5]: Yes, we certainly should.

**Pull request merged**